Refactor TextDecorationLineFlags into TextDecorationLine::Flag

vitorroriz opened the pull request:
#### a1433b9ad2b2a4e2f50d6f064db5642fe29b0cfa
<pre>
Refactor TextDecorationLineFlags into TextDecorationLine::Flag
<a href="https://rdar.apple.com/159463269">rdar://159463269</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=298115">https://bugs.webkit.org/show_bug.cgi?id=298115</a>

Reviewed by Tim Nguyen and Brent Fulgham.

Since TextDecorationLine is now converted to its own strong type,
it makes sense to refactor TextDecorationLineFlags into a nested
enum class &quot;Flag&quot; under TextDecorationLine.

Note that we also changed it from &quot;Flags&quot; to &quot;Flag&quot;,
as per Sammy&apos;s feedback at <a href="https://github.com/WebKit/WebKit/pull/49911">https://github.com/WebKit/WebKit/pull/49911</a>

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::LineDecorationStyle::LineDecorationStyle):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::logicalBottomForTextDecorationContent):
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::isDecoratingBoxForBackground):
* Source/WebCore/rendering/TextDecorationPainter.cpp:
(WebCore::TextDecorationPainter::paintBackgroundDecorations):
(WebCore::collectStylesForRenderer):
(WebCore::TextDecorationPainter::textDecorationsInEffectForStyle):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/svg/SVGTextBoxPainter.cpp:
(WebCore::SVGTextBoxPainter&lt;TextBoxPath&gt;::paint):
(WebCore::positionOffsetForDecoration):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.cpp:
(WebCore::operator&lt;&lt;):
(WebCore::Style::TextDecorationLine::addOrReplaceIfNotNone):
(WebCore::Style::CSSValueConversion&lt;TextDecorationLine&gt;::operator):
(WebCore::Style::CSSValueCreation&lt;OptionSet&lt;TextDecorationLine::Flag&gt;&gt;::operator):
(WebCore::Style::Serialize&lt;OptionSet&lt;TextDecorationLine::Flag&gt;&gt;::operator):
(WebCore::Style::operator&lt;&lt;):
(WebCore::Style::CSSValueCreation&lt;OptionSet&lt;TextDecorationLineFlags&gt;&gt;::operator): Deleted.
(WebCore::Style::Serialize&lt;OptionSet&lt;TextDecorationLineFlags&gt;&gt;::operator): Deleted.
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.h:
(WebCore::Style::TextDecorationLine::TextDecorationLine):
(WebCore::Style::TextDecorationLine::containsAny const):
(WebCore::Style::TextDecorationLine::contains const):
(WebCore::Style::TextDecorationLine::remove):
(WebCore::Style::TextDecorationLine::setFlags):
(WebCore::Style::TextDecorationLine::packFlags):
(WebCore::Style::TextDecorationLine::packFlagValue):
(WebCore::Style::TextDecorationLine::unpackFlags const):

Canonical link: <a href="https://commits.webkit.org/299349@main">https://commits.webkit.org/299349@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2f8cd1511b8e7b2b2600df1c62529874d340a335

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38347 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124846 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70726 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f49378db-3352-4337-86cc-dc522cb0a52b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120544 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39043 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46929 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90060 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7ec574f1-376b-4ddc-8799-09e4f6cdb117) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121619 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31104 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70566 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5f7c3cfc-63eb-4381-803d-000d2203e72b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30161 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/69094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100543 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127904 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45573 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34393 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98703 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45937 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98484 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25043 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43927 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/29551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42078 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45443 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51121 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44907 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48253 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46593 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
I don't like the name "Flag" but also don't have a much better name

brentfulgham reviewed (approved):
LGTM. r=me

vitorroriz committed:
Refactor TextDecorationLineFlags into TextDecorationLine::Flag
rdar://159463269
https://bugs.webkit.org/show_bug.cgi?id=298115

Reviewed by Tim Nguyen and Brent Fulgham.

Since TextDecorationLine is now converted to its own strong type,
it makes sense to refactor TextDecorationLineFlags into a nested
enum class "Flag" under TextDecorationLine.

Note that we also changed it from "Flags" to "Flag",
as per Sammy's feedback at https://github.com/WebKit/WebKit/pull/49911

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::LineDecorationStyle::LineDecorationStyle):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::logicalBottomForTextDecorationContent):
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::isDecoratingBoxForBackground):
* Source/WebCore/rendering/TextDecorationPainter.cpp:
(WebCore::TextDecorationPainter::paintBackgroundDecorations):
(WebCore::collectStylesForRenderer):
(WebCore::TextDecorationPainter::textDecorationsInEffectForStyle):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/svg/SVGTextBoxPainter.cpp:
(WebCore::SVGTextBoxPainter<TextBoxPath>::paint):
(WebCore::positionOffsetForDecoration):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.cpp:
(WebCore::operator<<):
(WebCore::Style::TextDecorationLine::addOrReplaceIfNotNone):
(WebCore::Style::CSSValueConversion<TextDecorationLine>::operator):
(WebCore::Style::CSSValueCreation<OptionSet<TextDecorationLine::Flag>>::operator):
(WebCore::Style::Serialize<OptionSet<TextDecorationLine::Flag>>::operator):
(WebCore::Style::operator<<):
(WebCore::Style::CSSValueCreation<OptionSet<TextDecorationLineFlags>>::operator): Deleted.
(WebCore::Style::Serialize<OptionSet<TextDecorationLineFlags>>::operator): Deleted.
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.h:
(WebCore::Style::TextDecorationLine::TextDecorationLine):
(WebCore::Style::TextDecorationLine::containsAny const):
(WebCore::Style::TextDecorationLine::contains const):
(WebCore::Style::TextDecorationLine::remove):
(WebCore::Style::TextDecorationLine::setFlags):
(WebCore::Style::TextDecorationLine::packFlags):
(WebCore::Style::TextDecorationLine::packFlagValue):
(WebCore::Style::TextDecorationLine::unpackFlags const):

Canonical link: https://commits.webkit.org/299349@main
